w3d_translator/classes/vector.py

```diff
@@ -0,0 +1,38 @@
+from dataclasses import dataclass, field
+
+from classes import XmlDataClass
+from utils import str_to_xy, str_to_xyz
+
+
+@dataclass
+class Vec2(XmlDataClass):
+    xy: dict = field(init=False)
+    x: float = field(init=False)
+    y: float = field(init=False)
+
+    def __init__(self, text):
+        super().__init__()
+        self.xy = str_to_xy(text)
```

Comment:
in the other constructors, you tend to use the pattern of having the initial item in the data, then transforming that in `__post_init__` - why not follow that same pattern here?

Reply:
is the `xy` string really needed separate from the `x` and `y`? It feels a little redundant to me

Reply:
I don't really see the value in storing the original string but that did get me thinking that I'm not entirely sure we need the original xml data for any of these classes? What do you think? 

I would just move to the `__init__()` pattern as done here. I definitely think storying the xml means passing around a lot of extra data and I don't have a reason for storing the data as of yet

Reply:
If we change over to that, then we're not really taking advantage of `DataClass` at all.  Do you really need to parse these things at init at all? or can you just use the class to "tag" the type of the xml element, so that when you write the yaml using `to_yaml` you can use the classes for dispatch and just put logic you need in those functions as needed.  The full classes with properties I don't see the utility in yet

Reply:
The issue has to do with making references to each other. For example, a `<Group>` is really just an array of strings, each one referring to an `[Object].name`. Same thing for `<Placement>`, `<Sound>`, and several others. Not saying the classes are the best way to go but I do need to represent the data in python somehow before I can translate things into yaml. Plus the yaml files will need to pass some sort of reference id when I start building parent/sibling relationships.

I personally feel most comfortable working in an OOP way but let me know your thoughts

Reply:
Turns out dataclass has a `field(repr=False)` to remove the field when printing which I added to all the classes. The least I can do for now, I'll throw something on the calendar to chat data structures?